core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -197,4 +198,35 @@ private RelDataType struct(RelDataType...relDataTypes) {
     }
     return builder.build();
   }
+
+  private void compareTypesIgnoringNullability(
+      RelDataType type1, RelDataType type2, boolean expectedResult) {
+    String msg = String.format(Locale.ROOT,
+        "%s and %s are expected to be %s ignoring nullability.",
+        type1.getFullTypeString(),
+        type2.getFullTypeString(),
+        expectedResult ? "equal" : "unequal");
+
```

Comment:
`unequal ` is not a valid word.

Reply:
Changed to 'not equal'. Thanks.

Reply:
Another issue here is that **both** asserts have the same message, so it is hard to tell which fails.

I would suggest:

```java
    assertThat(
        "SqlTypeUtil.equalSansNullability(typeFactory, " + type1 + ", " + type2 + ")",
        SqlTypeUtil.equalSansNullability(f.typeFactory, type1, type2), is(expectedResult));
    assertThat(
        "SqlTypeUtil.equalSansNullability(" + type1 + ", " + type2 + ")",
        SqlTypeUtil.equalSansNullability(type1, type2), is(expectedResult));
```

Reply:
Sounds reasonable. Revised accordingly. Thank you.